src/DataViews/FunctionsDataView.cpp

```diff
@@ -248,10 +247,16 @@ void FunctionsDataView::DoFilter() {
 void FunctionsDataView::AddFunctions(
     std::vector<const orbit_client_data::FunctionInfo*> functions) {
   functions_.insert(functions_.end(), functions.begin(), functions.end());
-  indices_.resize(functions_.size());
-  for (size_t i = 0; i < indices_.size(); ++i) {
-    indices_[i] = i;
-  }
```

Comment:
Why the `indices_` is not updated any more when adding functions?

Reply:
I just saw the `OnDataChange` calls `DoFilter` to clear the indices.

Reply:
Because `OnDataChanged()` takes care of properly updating it. That's why it's not updated in `ClearFunctions` either, for example. And note that this code as is is kind of wrong, because it doesn't consider sorting and filtering (which is responsibility of `OnDataChanged()`), so might as well remove it.

Reply:
It's not related to the PR but I just realize that we might have problem with multi-threading. 

We do not have mutex for `functions_` and `indices_`. What if someone use `indices_` after `ClearFunctions` clears `functions_` and before `OnDataChange` clears `indices_`?
We haven't run into issues for now as `AddFunctions` and `ClearFunctions` are called from OrbitApp (i.e., the main thread).

Anton and I discussed this offline. We might consider mutex or thread id check in the future.  
For now it would be great if we add a comment to `AddFunctions` and `ClearFunctions` to explain this should be only called from the main thread.

Reply:
And `RemoveFunctionsOfModule` for that matter

Reply:
In general my policy is that nothing is thread-safe unless it's documented as being explicitly thread safe.
I added a comment. But I think that regardless of the measure we take (comments, mutex, thread id check--the latest being probably the best), this is something that should be applied across to all `DataView`s, as it's not enough to only do it for the methods that a specific subclass overrides.

Reply:
Sounds like "Dataview refactor" to me. Lets leave it for another day ;)

Reply:
Hehe unfortunately